src/ai/BaseAI.js

```diff
@@ -245,6 +256,8 @@ export const BaseAI = class {
       this.stats.runtime.msec += msec;
       this.stats.runtime.sec += msec / 1000;
 
+      this.signal.removeEventListener('abort', hardStopListener);
```

Comment:
if (this.signal) ...

or this.signal?.removeEventListener(...) // preferred if this syntax works, please check it